[fix bug 1221328] Update /firefox/android/ send to device widget.

jpetto opened the pull request.

alexgibson commented on media/js/firefox/android.js:
```diff
@@ -283,85 +288,21 @@
-        // reference modal wrapper for widget
-        $sendToDeviceModal = $('#send-to-modal-container');
+        // add hidden input to update basket id
+        $('#send-to-device-form').append('<input type="hidden" name="send-to-device-basket-id" value="android-embed">');
```
If `android-embed` is now the default basket ID for this form, would it be better in the macro itself as opposed to added here in JS? I'm assuming this is triggering a specific email too, in which case two questions come to mind:

1.) Is it translated into all the languages we need to make this default?
2.) If so, should we just make this email the default basket ID for android, instead of sticking with the name `android-embed`? Would this be too much of a pain to make renaming worthwhile?

jpetto replied:
As per [ADavis' comment in the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1221328#c7), we want to keep the unique basket ID for this page.

However, your question regarding translation is still **very** valid. [Asked in the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1221328#c8). Marking as do not merge until we have a resolution.

alexgibson commented:
Can we also get rid of the variant logic in the android view? https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/views.py#L656

jpetto commented:
Hm, yeah, I think we can get rid of the variant logic for now. There is [another round of tests coming](https://bugzilla.mozilla.org/show_bug.cgi?id=1221338), but easy enough to go into the history to re-animate when they're ready.

jpetto commented:
[bniolet](https://bugzilla.mozilla.org/show_bug.cgi?id=1221328#c27) and [ADavis](https://bugzilla.mozilla.org/show_bug.cgi?id=1221328#c22) have given the green light in the bug, so we're good for final review and merge.

alexgibson commented:
Sorry for missing this first time around, but looks like the widget is off center on small viewports:

![mobile](https://cloud.githubusercontent.com/assets/400117/11928240/cb19bdfa-a7c9-11e5-814d-69d7aae5e167.png)

alexgibson commented on media/js/firefox/android.js:
```diff
@@ -23,6 +23,11 @@
     };
 
+    // send to device vars
+    var hasWidget = $('#intro').data('has-widget') !== 'False';
```
Nit: even though jQuery is relax on data-attributes, I believe the standards way is to use camelCase: `$('#intro').data('hasWidget')`

jpetto replied:
Ah, yes indeed cameCase would be better. :+1:

alexgibson commented:
r+wc from me :frog:

jpetto commented:
Fixed tablet/mobile centering and updated jQuery data attribute format.

Agree that a unique basket id on this page complicates things. I believe it's been decided that the increased conversions are worth the dev cost, but I'd like to keep this on the radar as something to possibly improve in 2016.
